src/System.IO.FileSystem/tests/Directory/EnumerableTests.cs

```diff
@@ -32,6 +32,32 @@ public void FileEnumeratorIsThreadSafe()
             }
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void EnumerateDirectories_NonBreakingSpace()
+        {
+            var expectedDirectoryNames = new List<string> { GetTestFileName(), "\u00A0", GetTestFileName() };
+
+            DirectoryInfo di = new DirectoryInfo(Path.Combine(TestDirectory, expectedDirectoryNames[0]));
+            di.Create();
+            di.CreateSubdirectory(expectedDirectoryNames[1]);
+            di.CreateSubdirectory(expectedDirectoryNames[1] + Path.DirectorySeparatorChar + expectedDirectoryNames[2]);
+            
+            var actualDirectoryNames = new List<string>();
+            AddDirectoryNameToList(di, actualDirectoryNames);
+
+            Assert.Equal(expectedDirectoryNames, actualDirectoryNames);
+        }
+
+        public void AddDirectoryNameToList(DirectoryInfo di, List<string> directoryNames)
```

Comment:
You could tidy this up by embedding it as a [Local function](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#local-functions)